presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/AbstractCostBasedPlanTest.java

```diff
@@ -0,0 +1,234 @@
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.sql.planner.plan.JoinNode.DistributionType.REPLICATED;
+import static com.facebook.presto.sql.planner.plan.JoinNode.Type.INNER;
+import static com.facebook.presto.testing.TestngUtils.toDataProvider;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.io.MoreFiles.getNameWithoutExtension;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.Files.isDirectory;
+import static java.util.Locale.ENGLISH;
+import static java.util.stream.Collectors.joining;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractCostBasedPlanTest
+        extends BasePlanTest
+{
+    public AbstractCostBasedPlanTest(LocalQueryRunnerSupplier supplier)
+    {
+        super(supplier);
+    }
+
+    protected abstract Stream<Path> getQueryResourcePaths();
+
+    @DataProvider
+    public Object[][] getQueriesDataProvider()
+    {
+        return getQueryResourcePaths()
+                .collect(toDataProvider());
+    }
+
+    @Test(dataProvider = "getQueriesDataProvider")
+    public void test(Path queryResourcePath)
+    {
+        assertEquals(generateQueryPlan(read(queryResourcePath)), read(getQueryPlanResourcePath(queryResourcePath)));
+    }
+
+    private Path getQueryPlanResourcePath(Path queryResourcePath)
+    {
+        return Paths.get(queryResourcePath.getParent().toString(), getNameWithoutExtension(queryResourcePath) + ".plan.txt");
+    }
+
+    public void generate()
+            throws Exception
+    {
+        initPlanTest();
+        try {
+            getQueryResourcePaths()
+                    .parallel()
+                    .forEach(queryResourcePath -> {
+                        try {
+                            Path queryPlanWritePath = Paths.get(
+                                    getPrestoMainSourcePath().toString(),
+                                    "src/test/resources",
+                                    getQueryPlanResourcePath(queryResourcePath).toString());
+                            Files.createParentDirs(queryPlanWritePath.toFile());
+                            Files.write(generateQueryPlan(read(queryResourcePath)).getBytes(UTF_8), queryPlanWritePath.toFile());
+                            System.out.println("Generated expected plan for query: " + queryResourcePath);
+                        }
+                        catch (IOException e) {
+                            throw new UncheckedIOException(e);
+                        }
+                    });
+        }
+        finally {
+            destroyPlanTest();
+        }
+    }
+
+    private static String read(Path resource)
+    {
+        try {
+            return Resources.toString(Resources.getResource(AbstractCostBasedPlanTest.class, resource.toString()), UTF_8);
```

Comment:
static import `getResource`. Also you can use `getResource(string)`. Context class loader should be fine here.

Reply:
> Context class loader should be fine here.

It does not work.